homeassistant/components/zwave/__init__.py

```diff
@@ -752,24 +605,166 @@ def start_zwave(_service_or_event):
+                continue
+            self._values[name] = value
+            if self._entity:
+                self._entity.value_changed()
+
+            self._check_entity_ready()
+
+    def _check_entity_ready(self):
+        """Check if all required values are discovered and create entity."""
+        if self._workaround_ignore:
+            return
+        if self._entity is not None:
+            return
+
+        for name in self._schema[const.DISC_VALUES]:
+            if self._values[name] is None and \
+                    not self._schema[const.DISC_VALUES][name].get(
+                            const.DISC_OPTIONAL):
+                return
+
+        component = self._schema[const.DISC_COMPONENT]
+
+        workaround_component = workaround.get_device_component_mapping(
+            self.primary)
+        if workaround_component and workaround_component != component:
+            if workaround_component == workaround.WORKAROUND_IGNORE:
+                _LOGGER.info("Ignoring device %s due to workaround.",
+                             "{}.{}".format(
+                                 component, object_id(self.primary)))
+                # No entity will be created for this value
+                self._workaround_ignore = True
+                return
+            _LOGGER.debug("Using %s instead of %s",
+                          workaround_component, component)
+            component = workaround_component
+
+        name = "{}.{}".format(component, object_id(self.primary))
+        node_config = self._device_config.get(name)
+
+        # Configure node
+        _LOGGER.debug("Adding Node_id=%s Generic_command_class=%s, "
+                      "Specific_command_class=%s, "
+                      "Command_class=%s, Value type=%s, "
+                      "Genre=%s as %s", self._node.node_id,
+                      self._node.generic, self._node.specific,
+                      self.primary.command_class, self.primary.type,
+                      self.primary.genre, component)
+
+        if node_config.get(CONF_IGNORED):
+            _LOGGER.info(
+                "Ignoring node %s due to device settings.", self._node.node_id)
+            # No entity will be created for this value
+            self._workaround_ignore = True
+            return
+
+        polling_intensity = convert(
+            node_config.get(CONF_POLLING_INTENSITY), int)
+        if polling_intensity:
+            self.primary.enable_poll(polling_intensity)
+        else:
+            self.primary.disable_poll()
+
+        platform = get_platform(component, DOMAIN)
+        device = platform.get_device(
+            node=self._node, values=self,
+            node_config=node_config, hass=self._hass)
+        if not device:
+            # No entity will be created for this value
+            self._workaround_ignore = True
+            return
+
+        self._entity = device
+
+        dict_id = "{}.{}".format(component, self.primary.value_id)
```

Comment:
Is it possible that this is not unique? i.e. that there is more than one discovery for the same primary value using the same component? Maybe add / replace it with `id(self)` ?